osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCounterDisplay.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Screens.Play.HUD.JudgementCounter
+{
+    public partial class JudgementCounterDisplay : CompositeDrawable, ISkinnableDrawable
+    {
+        public bool UsesFixedAnchor { get; set; }
+
+        [SettingSource("Display mode")]
+        public Bindable<DisplayMode> Mode { get; set; } = new Bindable<DisplayMode>();
+
+        [SettingSource("Counter direction")]
+        public Bindable<Flow> FlowDirection { get; set; } = new Bindable<Flow>();
+
+        [SettingSource("Show judgement names")]
+        public BindableBool ShowName { get; set; } = new BindableBool(true);
+
+        [SettingSource("Show max judgement")]
+        public BindableBool ShowMax { get; set; } = new BindableBool(true);
+
+        [Resolved]
+        private JudgementTally tally { get; set; } = null!;
+
+        protected FillFlowContainer JudgementContainer;
+
+        public JudgementCounterDisplay()
+        {
+            AutoSizeAxes = Axes.Both;
+            InternalChild = JudgementContainer = new FillFlowContainer
+            {
+                Direction = getFlow(FlowDirection.Value),
+                Spacing = new Vector2(10),
+                AutoSizeAxes = Axes.Both
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            //Adding this in "load" will cause the component to not load in properly after the first beatmap attempt. Or after existing and reentering.
```

Comment:
```suggestion
            //Adding this in "load" will cause the component to not load in properly after the first beatmap attempt. Or after exiting and reentering.
```

This comment is pointless unless you include the "why". If you haven't figured out the "why" yet then it needs to be understood before this can be merged.

Reply:
Took quite a bit longer than I'd like to admit ( i was rather baffled by the apparent randomness of the issue) , but I think I found the "why".

HUDOverlay creates both the JudgementTally and the JudgementCounter and since the tally was being assigned at the bottom of the HUDoverlays "CTOR" it was often not ready in time for when the counter wanted to fetch the info. From what I've tested  two solutions avoid the issue entirely, loading it at the top of the HUDOverlay's children ( line 109) before the "mainComponents" container , or keeping the "for each" loop in the load complete as i have it here, which would be preferable ?